### PR TITLE
Add GZip support

### PR DESCRIPTION
**Pull request by tculotta**

### CODE REVIEW

File: ExampleTestProject/Assets/PlayFabSDK/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs [hunk 1]
```diff
@@ -43,6 +43,16 @@ private static IEnumerator SimpleCallCoroutine(string method, string fullUrl, by
             {
                 using (UnityWebRequest www = UnityWebRequest.Get(fullUrl))
                 {
+                    if (PlayFabSettings.staticSettings.CompressResponses)
```
Comment (amccalib) [inline ExampleTestProject/Assets/PlayFabSDK/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs:46]: One thought here. In the C# version we allow the CompressResponses option to be set and honored at an API class instance scope. This looks like it's just honoring the global scope. That was easy to do in the C# version, but I'm not sure it's as easy here.

File: ExampleTestProject/Assets/PlayFabSDK/Shared/Models/PlayFabSharedSettings.cs [hunk 2]
```diff
@@ -25,6 +25,8 @@ public class PlayFabSharedSettings : ScriptableObject
 
     public int RequestTimeout = 2000;
     public bool RequestKeepAlive = true;
+    public bool CompressResponses = false;
+    public bool DecompressWithDownloadHandler = true;
```
Comment (amccalib) [inline ExampleTestProject/Assets/PlayFabSDK/Shared/Models/PlayFabSharedSettings.cs:29]: It's not clear to me why you'd ever set this to false. Maybe you explained that elsewhere and I didn't follow.

File: ExampleTestProject/Assets/PlayFabSDK/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs (hunk 1 above)
Comment (tculotta) [reply ExampleTestProject/Assets/PlayFabSDK/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs:46]: I’ll take another look at the C# SDK and see if what I might be able to do like that here. Definitely going to be harder I think

File: ExampleTestProject/Assets/PlayFabSDK/Shared/Models/PlayFabSharedSettings.cs (hunk 2 above)
Comment (tculotta) [reply ExampleTestProject/Assets/PlayFabSDK/Shared/Models/PlayFabSharedSettings.cs:29]: This is what I brought up in our teams chat and copied to the thread in the code review channel as well. I can copy that note over to here too though.

Comment (tculotta) [reply ExampleTestProject/Assets/PlayFabSDK/Shared/Models/PlayFabSharedSettings.cs:29]: There’s also a comment about it within the GZipDownloadHandler

**Pull request closed without merging**